src/role-resolver.js

```diff
@@ -0,0 +1,70 @@
+angular.module('mmRoute')
+.factory('mmRoleResolver', function ( $window, $injector, mmRoute ) {
+
+  function checkTemplatePermission( accessRole ) {
+    var roles = $injector.invoke(mmRoute.roleGetter, $window);
+
+    if ( !Array.isArray(accessRole) ) {
+      return roles.indexOf(accessRole) !== -1;
+    } else {
+      return accessRole.every(function ( singleRole ) {
+        return roles.indexOf(singleRole) !== -1;
+      });
+    }
+  }
+
+  function chooseTemplate( accessObjects ) {
+    var chosenTemplate = null;
+
+    for ( var i = 0; i < accessObjects.length; i++ ) {
+      var accessObject = accessObjects[ i ];
+      if ( accessObject.roles.some(checkTemplatePermission) ) {
+        chosenTemplate = accessObject.page;
+        break;
+      }
+    }
+
+    if ( !chosenTemplate ) {
+      throw new Error();
+    }
+
+    return chosenTemplate;
+
+  }
+
+  return {
+    chooseTemplate: chooseTemplate,
+    _checkTemplatePermission: checkTemplatePermission
+  };
+
+})
+.directive('mmRoleResolver', function ( $http, $compile, $templateCache, $controller, $location, mmRoute, mmRoleResolver  ) {
+  return {
+    restrict: 'E',
+    scope: {
+      route: '=',
+    },
+    link: function ( scope, element ) {
+      var chosenTemplate;
+      try {
+        chosenTemplate = mmRoleResolver.chooseTemplate(scope.route.access);
+      }
+      catch ( err ) {
+        $location.path('/404');
```

Comment:
This should redirect to whatever is configured with the provider as the default route.